python/pyiceberg/schema.py

```diff
@@ -317,6 +331,97 @@ def primitive(self, primitive: PrimitiveType) -> T:
         """Visit a PrimitiveType"""
 
 
+class SchemaVisitorPerPrimitiveType(SchemaVisitor[T], ABC):
+    def primitive(self, primitive: PrimitiveType) -> T:
+        """Visit a PrimitiveType"""
+        if isinstance(primitive, FixedType):
```

Comment:
Are repeated `isinstance` checks going to introduce overhead compared to using `@singledistpatchmethod` here?

Reply:
It should be roughly equivalent, the `isinstance` checks are not that expensive. The main reason to go this route is that we can just OOP to call the right method, and you can't override a `singledistpatchmethod` method, it will still call the one on the base class.